resources/packs/ms365/msgraph.go

```diff
@@ -9,17 +9,32 @@ import (
 	"github.com/microsoftgraph/msgraph-beta-sdk-go/models"
 	"github.com/microsoftgraph/msgraph-beta-sdk-go/rolemanagement/directory/roleassignments"
 	"go.mondoo.com/cnquery/motor/providers"
-	ms365_provider "go.mondoo.com/cnquery/motor/providers/ms365"
-	"go.mondoo.com/cnquery/motor/providers/ms365/msgraphclient"
-	"go.mondoo.com/cnquery/motor/providers/ms365/msgraphconv"
+	"go.mondoo.com/cnquery/motor/providers/microsoft/azure"
+	ms365_provider "go.mondoo.com/cnquery/motor/providers/microsoft/ms365"
+	"go.mondoo.com/cnquery/motor/providers/microsoft/ms365/msgraphclient"
+	"go.mondoo.com/cnquery/motor/providers/microsoft/ms365/msgraphconv"
+	"go.mondoo.com/cnquery/motor/vault"
 	"go.mondoo.com/cnquery/resources"
 	"go.mondoo.com/cnquery/resources/packs/core"
 )
 
-func ms365Provider(t providers.Instance) (*ms365_provider.Provider, error) {
+func msGraphProvider(t providers.Instance) (*ms365_provider.Provider, error) {
 	at, ok := t.(*ms365_provider.Provider)
 	if !ok {
-		return nil, errors.New("ms365 resource is not supported on this transport")
+		az, ok := t.(*azure.Provider)
+		if !ok {
+			return nil, errors.New("msgraph resource is not supported on this transport")
+		}
+		cred := az.Credential()
+		if cred == nil {
+			return nil, errors.New("msgraph resource requires credentials")
+		}
+		cfg := &providers.Config{
```

Comment:
Note: this is a temporary bridge to allow running `msgraph` queries onto the azure provider. Ideally, we don't want to create a new provider for each query here